### PR TITLE
Deferred batches

### PR DESCRIPTION
**Pull request by jherico**
Part 1 of a project to migrate backend rendering execution off the main thread.  This PR will cause all batches to be generated for a frame, and once the frame is complete, cause all the batches to be executed, rather than doing a per-batch build/execute

## Testing

There should be no significant impact on performance or behavior of the application.  Verify that mini-mirror, and all camera views continue to function as before.

### CODE REVIEW

**Comment by jherico** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;SkO1DPvx&#x2F;5-modify-gpu-context-to-include-an-end-frame-api"> Modify gpu::Context to include an end-frame API</a>

**Comment by jherico** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;fBzxJzvk&#x2F;10-have-gpu-context-accumulate-batches-and-then-execute-them-through-a-handler-in-the-end-frame-call"> Have gpu::Context accumulate batches and then execute them through a handler in the end frame call</a>

**Comment by jherico** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;fKASnTDc&#x2F;9-modify-application-to-use-the-new-context-api"> Modify application to use the new Context API</a>

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1711,10 +1722,6 @@ void Application::paintGL() {
         QSize size = getDeviceSize();
         renderArgs._viewport = glm::ivec4(0, 0, size.width(), size.height());
         _applicationOverlay.renderOverlay(&renderArgs);
-        auto overlayTexture = _applicationOverlay.acquireOverlay();
```
Comment (jherico) [inline interface/src/Application.cpp]: `acquireOverlay` must happen after the execution of the batch, so it's moved to after frame completion.

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -1900,11 +1910,6 @@ void Application::paintGL() {
 
     {
         Stats::getInstance()->setRenderDetails(renderArgs._details);
-        // Reset the gpu::Context Stages
-        // Back to the default framebuffer;
-        gpu::doInBatch(renderArgs._context, [&](gpu::Batch& batch) {
```
Comment (jherico) [inline interface/src/Application.cpp]: Must happen inside the frame definition, so moved to just after the `beginFrame` call

**Comment by hifi-gustavo** [comment]
Build console output for commit: a9740b803fa91931e845615a8fbf5efad2c5f54d available at: https://gist.github.com/fa8ff84892036515c0cd787e3b3618c3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8322-a9740b8.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8322-a9740b8.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 54a7265be8ee66b3a4f0a849ccb70c9ef7fb8edd available at: https://gist.github.com/fa8ff84892036515c0cd787e3b3618c3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8322-54a7265.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8322-54a7265.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5bbcbbb811985e1bfba7bd054f95ec72f82f1739 available at: https://gist.github.com/fa8ff84892036515c0cd787e3b3618c3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8322-5bbcbbb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8322-5bbcbbb.dmg

**Comment by thoys** [comment]
# QA Testing report

All the different view modes work fine, got 60FPS desktop on playa, and 90FPS in vive.

QA Pass

**Pull request merged**